example/src/new_api/swipeable/Swipeable.tsx

```diff
@@ -0,0 +1,606 @@
+   */
+  onSwipeableOpenStartDrag?: (direction: 'left' | 'right') => void;
+
+  /**
+   * Called when action panel starts being shown on dragging to close.
+   */
+  onSwipeableCloseStartDrag?: (direction: 'left' | 'right') => void;
+
+  /**
+   *
+   * This map describes the values to use as inputRange for extra interpolation:
+   * AnimatedValue: [startValue, endValue]
+   *
+   * progressAnimatedValue: [0, 1] dragAnimatedValue: [0, +]
+   *
+   * To support `rtl` flexbox layouts use `flexDirection` styling.
+   * */
+  renderLeftActions?: (
+    progressAnimatedValue: SharedValue,
+    dragAnimatedValue: SharedValue,
+    swipeable: ForwardedRef<SwipeableProps & RefAttributes<ExposedFunctions>> // we have to use ref here, as it now holds all the objects
+  ) => React.ReactNode;
+  /**
+   *
+   * This map describes the values to use as inputRange for extra interpolation:
+   * AnimatedValue: [startValue, endValue]
+   *
+   * progressAnimatedValue: [0, 1] dragAnimatedValue: [0, -]
+   *
+   * To support `rtl` flexbox layouts use `flexDirection` styling.
+   * */
+  renderRightActions?: (
+    progressAnimatedValue: SharedValue,
+    dragAnimatedValue: SharedValue,
+    swipeable: ForwardedRef<SwipeableProps & RefAttributes<ExposedFunctions>>
+  ) => React.ReactNode;
+
+  useNativeAnimations?: boolean;
+
+  animationOptions?: Record<string, unknown>;
+
+  /**
+   * Style object for the container (`Animated.View`), for example to override
+   * `overflow: 'hidden'`.
+   */
+  containerStyle?: StyleProp<ViewStyle>;
+
+  /**
+   * Style object for the children container (`Animated.View`), for example to
+   * apply `flex: 1`
+   */
+  childrenContainerStyle?: StyleProp<ViewStyle>;
+}
+
+export interface ExposedFunctions {
+  close: () => void;
+  openLeft: () => void;
+  openRight: () => void;
+  reset: () => void;
+}
+
+const Swipeable = forwardRef<
+  SwipeableProps & React.RefAttributes<ExposedFunctions>
+>(function Swipeable(
+  props: SwipeableProps,
+  ref: ForwardedRef<SwipeableProps & RefAttributes<ExposedFunctions>>
+) {
+  const [rowState, setRowState] = useState<number>(0);
+
+  const dragX = useSharedValue<number>(0);
+  const rowTranslation = useSharedValue<number>(0);
+  const leftWidth = useSharedValue<number>(0);
+  const rightOffset = useSharedValue<number>(0);
+  const rowWidth = useSharedValue<number>(0);
+
+  // todo: if things don't work, set all the default values from the original file
+  // they are set everywhere thoughout this file, many functions set the default states themselves
+
+  const defaultProps = {
+    friction: 1,
+    overshootFriction: 1,
+    useNativeAnimations: true,
+  };
+
+  useEffect(() => {
+    if (!props.friction) props.friction = defaultProps.friction;
+    if (!props.overshootFriction)
+      props.overshootFriction = defaultProps.overshootFriction;
+    if (!props.useNativeAnimations)
+      props.useNativeAnimations = defaultProps.useNativeAnimations;
+
+    updateAnimatedEvent();
+  }, []);
```

Comment:
I don't think this is the right to do. Reassigning prop values is an anti-pattern, instead, you can use something like this:
```js
const { foo = 'default' } = props;
```

and either destructure the props at the top level and use those values in the code, or move default values to global constants and use them where necessary. Personally I'd lean more towards the first approach.